lib/cgus.py

```diff
@@ -0,0 +1,62 @@
+import json
+import logging
+from pathlib import Path, PosixPath
+
+import nltk
+
+logging.basicConfig(level=logging.INFO)
+
+
+class CGUsDataset():
+    """
+    Helper class for handling CGUs Versions
+    """
+
+    def __init__(self, root_path: str = "../CGUs-versions"):
+        self.root_path = Path(root_path)
+        assert self.root_path.exists(), f"{root_path} does not exist"
+        assert self.root_path.is_dir(), f"{root_path} is not a directory"
+
+
+    def yield_all_md(self, ignore_rootdir: bool = True) -> list:
+        if ignore_rootdir:
+            return self.root_path.glob('**/*/*.md')
+        else:
+            return self.root_path.glob('**/*.md')
+
+class CGU():
+    """
+    A CGU object, and the main building bloc for our algorithms.
+    """
+
+    # Remove all punctuation when tokenizing text
+    _TOKENIZER = nltk.tokenize.RegexpTokenizer(r'\w+')
+
+    def __init__(self, path: PosixPath):
+        self._path = path
+        self.name = self._path.name
+        self.source = self._path.as_posix().split("/")[-2]
```

Comment:
I think here `source` is the `service` name. I suggest to keep same terms between the two projects.